manifests/config.pp

```diff
@@ -15,6 +15,13 @@
 # -----------------------------------------------------------------------------
 class jira::config inherits jira {
 
+  if $jira::data_dir {
```

Comment:
This is really confusing. You might want to consider using the ```$_data_dir``` convention to make it clear that this variable is not used in any class elsewhere